stock_quant_history/models/stock_quant_history_snapshot.py

```diff
@@ -151,38 +151,28 @@ def _generate_stock_quant_history(self):
         _logger.info(
             "Apply %s stock.move.line since previous snapshot", len(stock_move_lines)
         )
+        pickings = self.env["stock.picking"]
         ignored_location_usage = self._ignored_location_usage()
         for move_line in stock_move_lines:
+            quant_copy = quant_history[
+                (move_line.product_id, move_line.lot_id, move_line.location_id)
```

Comment:
surprised that test do not failed probably we can improve but depending of the condition se use the location_id or location_dest_id